public/main.js

```diff
@@ -52,9 +52,30 @@ function reloadTodoList() {
         todos.forEach(function(todo) {
             var listItem = document.createElement("li");
             listItem.textContent = todo.title;
+            var deleteButton = document.createElement("button");
+            deleteButton.textContent = "delete";
+            deleteButton.className = "deleteButton";
+            deleteButton.addEventListener("click", function() {
+                deleteEntry(todo);
+            });
+            listItem.appendChild(deleteButton);
             todoList.appendChild(listItem);
         });
     });
 }
 
+function deleteEntry(todo) {
+    var createRequest = new XMLHttpRequest();
+    createRequest.open("DELETE", "/api/todo/" + todo.id);
+    createRequest.setRequestHeader("Content-type", "application/json");
+    createRequest.onload = function() {
```

Comment:
This could look like:
```
function() {
    if(this.state !== 200) {
        error.textContent = "Failed to delete. Server returned " + this.status + " - " + this.responseText;

        return;
    }

    reloadTodoList();
}
```

Helps to save nesting nightmares potentially. I made the error first, because it's good to get them out of the way and with some callbacks, you'll find them first. This is just a suggestion though, so feel free to leave it if you want